### PR TITLE
[GA] Enable Kyuubi nightly build against Spark master

### PR DESCRIPTION
**Pull request by pan3793**
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+name: Kyuubi Nightly
+
+on:
+  schedule:
+    - cron: 0 23 * * *
```
Comment (yaooqinn) [inline .github/workflows/nightly.yml:5]: What is the time for spark nightly build?

**Review by yaooqinn — APPROVED** [review]

File: .github/workflows/nightly.yml (hunk 1 above)
Comment (pan3793) [reply .github/workflows/nightly.yml:5]: midnight, and should we change it to 1:00 am?

Comment (pan3793) [reply .github/workflows/nightly.yml:5]: Have look on Spark GA list, nightly build usually cost 1~2 hours, and sometime over 3 hours, maybe we should start at 4:00 am.

**Comment by yaooqinn** [comment]
will merge this affer 6:00 pm, Asia/Shanghai

**Comment by yaooqinn** [comment]
thanks, merged to master for v1.3.0
